cinema/views.py

```diff
@@ -56,3 +88,50 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related("movie", "cinema_hall")
+                .annotate(
+                    tickets_available=(
+                        F("cinema_hall__rows") * F("cinema_hall__seats_in_row")
+                    ) - Count("tickets")
+                )
+            )
+            movie = self.request.query_params.get("movie")
+            date = self.request.query_params.get("date")
+            if movie:
+                queryset = queryset.filter(movie__id=movie)
+
+            if date:
+                queryset = queryset.filter(show_time__date=date)
+
+        if self.action == "retrieve":
+            queryset = queryset.select_related("movie", "cinema_hall")
+
+        return queryset.distinct()
+
+
+class OrdersViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    pagination_class = OrderSetPagination
+
+    def get_queryset(self):
+        queryset = self.queryset.filter(user=self.request.user)
+        if self.action == "list":
+            queryset = queryset.prefetch_related("tickets__movie_session")
```

Comment:
here would be better to specifically prefetch "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"